tutorbackup/templates/backup/build/backup/restore_services.py

```diff
@@ -121,11 +121,17 @@ def download_from_s3(version_id=None):
         )
 
         logger.info("Checking downloaded file's integrity ...")
-        obj_metadata = S3_CLIENT.head_object(
-            Bucket=bucket,
-            Key=os.path.basename(file_name),
-            VersionId=version_id,
-        )
+        if version_id:
```

Comment:
You may want to add a comment here why this is necessary, to deter a future contributor (including future Foad : ) ) from unduly "simplifying" this bit.

Reply:
That's a good idea. I'll do that